File: DSA/src/datastructures/nodes/single/SingleNode.java
package datastructures.nodes.single;

public class SingleNode<T> {
    private T data;
    private SingleNode<T> next;

    public SingleNode() {
        this.data = null;
        this.next = null;
    }

    public SingleNode(T data, SingleNode next) {
        this.data = data;
        this.next = next;
    }

    public T getData() {
        return data;
    }

    public void setData(T data) {
        this.data = data;
    }

    public SingleNode<T> getNext() {
        return next;
    }

    public void setNext(SingleNode<T> next) {
        this.next = next;
    }
}

File: DSA/DataStructures/Nodes/SingleNodeNS/SingleNode.cs
namespace DSA.DataStructures.Nodes.SingleNodeNS
{
  public class SingleNode<T>
  {
    public T? Data { get; set; }
    public SingleNode<T>? Next { get; set; }

    public SingleNode()
    {
      this.Data = default(T);
      this.Next = null;
    }

    public SingleNode(T data)
    {
      this.Data = data;
      this.Next = null;
    }

    public SingleNode(T data, SingleNode<T> node)
    {
      this.Data = data;
      this.Next = node;
    }
  }
}